packages/app/src/cli/services/generate/shop-import/declarative-definitions.ts

```diff
@@ -0,0 +1,742 @@
+            after: cursor,
+          }),
+        })
+        metafieldLoadResults.push({
+          metafields,
+          ownerType: dcddOwner as MetafieldOwners,
+          graphQLOwner,
+        })
+      },
+    })),
+  )
+
+  const metaobjects = await renderSingleTask({
+    title: outputContent`Loading ${outputToken.green('metaobjects')}`,
+    task: async () => {
+      return paginatedQuery({
+        query: MetaobjectDefinitions,
+        session: adminSession,
+        toNodes: (res) => res.metaobjectDefinitions,
+        toVariables: (cursor) => ({
+          after: cursor,
+        }),
+      })
+    },
+  })
+
+  // Prepare inputs for processing
+  const metafieldNodes: MetafieldNodesInput[] = []
+  for (const {metafields, ownerType, graphQLOwner} of metafieldLoadResults) {
+    if (metafields.status === 'ok') {
+      metafieldNodes.push({
+        ownerType,
+        items: metafields.items,
+        graphQLOwner,
+      })
+    }
+  }
+
+  const metaobjectNodes: MetaobjectForImportFragment[] = metaobjects.status === 'ok' ? metaobjects.items : []
+
+  // Process all nodes
+  const {tomlContent, metafieldCount, metaobjectCount} = processDeclarativeDefinitionNodes(
+    metafieldNodes,
+    metaobjectNodes,
+  )
+
+  renderInfo({
+    headline: 'Conversion to TOML complete.',
+    body: [
+      'Converted',
+      {
+        warn: `${metafieldCount} metafields`,
+      },
+      'and',
+      {
+        warn: `${metaobjectCount} metaobjects`,
+      },
+      'from',
+      {
+        warn: shopName,
+      },
+      'into TOML, ready for you to copy.',
+    ],
+    orderedNextSteps: true,
+    nextSteps: [
+      'Review the suggested TOML carefully before applying.',
+      [
+        'Missing sections? Make sure your app has the required access scopes to load metafields and metaobjects (e.g.',
+        {
+          command: 'read_customers',
+        },
+        'to load customer metafields,',
+        {
+          command: 'read_metaobject_definitions',
+        },
+        'to load metaobjects.)',
+      ],
+      [
+        'Missing definitions? Only metafields and metaobjects that are app-reserved (using',
+        {
+          command: '$app',
+        },
+        ') will be converted.',
+      ],
+      [
+        "When you're ready, add the generated TOML to your app's configuration file and test out changes with the",
+        {
+          command: 'shopify app dev',
+        },
+        'command.',
+      ],
+    ],
+  })
+
+  renderTomlStringWithFormatting(tomlContent)
+}
+
+type ConvertedMetafield =
+  | {
+      status: 'ok'
+      metafield: Metafield
+      namespace: string
+      key: string
+      patches: Patch[]
+    }
+  | {
+      status: 'not_app_reserved'
+    }
+
+async function createAdminApiSessionForShop(options: ImportDeclarativeDefinitionsOptions) {
+  const {app, remoteApp, developerPlatformClient, organization, specifications} = options
+
+  const store = await storeContext({
```

Comment:
I think we should move this out of this file and just pass the store, it's quite unrelated